rust/core-lib/src/line_ending.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate xi_rope;
+
+use xi_rope::Rope;
+use memchr::memchr;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum LineEnding {
+    Crlf,
```

Comment:
CrLf, maybe, although I missed the bikeshedding so I'm happy to be ignored. It might be worth adding a brief comment to each of these for readers who aren't familiar with the notation. (And my real vote might indeed be for `UnixStyle` and `DosStyle` or something. But really, I'm okay with whatever)

Reply:
Hmm... I CrLf would make sense because it's essentially two separate acronyms. It might also be slightly easier to read.